### ISSUE TITLE
React warning about unrecognized prop `dataIconPath` on DOM element in InlineNotification component

### ISSUE THREAD

**Issue by austinbruch** [IBM/carbon-elements#498]
## Detailed description

> Describe in detail the issue you're having.

When rendering an `InlineNotification` component, React is giving a warning about an unrecognized prop on a `path` DOM element (see stack trace below).

> Is this issue related to a specific component?

Yes, `InlineNotification`.

> What did you expect to happen? What happened instead? What would you like to see changed?

Expected no warnings from React.

> What browser are you working in?

Chrome 73

> What version of the Carbon Design System are you using?

v7.1.0

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?

IBM Cloud Identity

## Steps to reproduce the issue

1. Render InlineNotification with minimum required props, specifying `kind` as `'success'`. Other kinds also cause the same warning.

> Please create a reduced test case in CodeSandbox

https://codesandbox.io/s/z318q1pp7l

## Additional information

- Screenshots or code
- Notes
react-dom.development.js:506 Warning: React does not recognize the `dataIconPath` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `dataiconpath` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
    in path (created by ErrorFilled20)
    in svg (created by ErrorFilled20)
    in ErrorFilled20 (created by NotificationIcon)
    in NotificationIcon (created by InlineNotification)
    in div (created by InlineNotification)
    in div (created by InlineNotification)

**Comment by AlanGreene** [comment]
Is there a release planned to include this fix? I see the last release is 25 days old. This warning is generating a lot of noise in tests and in our application.

**Comment by joshblack** [comment]
Hi @AlanGreene! 👋 This will be included in the next release of this project, for sure 👍

**Comment by ericlowry** [comment]
I am seeing the same behavior in `<TextInput invalid invalidText="Bad!" value="xxx">`, has this fix been released and I just missed it?

**Comment by ericlowry** [comment]
scratch that, I manually added @carbon/icons as a dependency in my project and it fixed it...

**Comment by joshblack** [comment]
Hi @ericlowry! 👋 The fix was just published in 10.2.0 and here's the PR to bring it into `carbon-components-react`: https://github.com/carbon-design-system/carbon/pull/2706

**Comment by ericlowry** [comment]
Hey Josh, thanks for the follow up - so I upgraded everything to 10.2.0 and now I am getting the error again :-(
```
  "dependencies": {
    "@carbon/icons": "^10.2.0",
    "@carbon/icons-react": "^10.2.0",
    "@carbon/themes": "^10.2.0",
    "apollo-boost": "^0.3.1",
    "apollo-client": "^2.5.1",
    "apollo-client-mock": "^0.0.9",
    "apollo-link-context": "^1.0.17",
    "carbon-components": "^10.2.0",
    "carbon-components-react": "^7.2.0",
...
```

Am I missing some step?

**Comment by joshblack** [comment]
Hi @ericlowry! Most likely the error is coming from `carbon-components-react` as we haven't published a patch release yet 👍

**Comment by xrissot-ibm** [comment]
Hi @joshblack any idea of the ETA for the new release with the fix?

**Comment by Leapfrognz** [comment]
When is this fix going to be released? it is breaking our app.

**Comment by joshblack** [comment]
Hi all! This will be included in our 7.3.0 release next week 👍
